### PR TITLE
Adjustments to fit the App

### PR DESCRIPTION
**Pull request by andped10**
General
- added `as_dict` method for classes where the produced dictionary needs to be customized to fit the constructor
- added pipelines to handle draft of release notes 

In `model.py` made it more clear that the underlying `sample` consists of `assemblies` 

In `surfactant_layer.py`
- protected `_area_per_molecule`

In `layer_area_per_molecule.py`
- protected `_area_per_molecule`
- protected `_scattering_length_real`
- protected `_scattering_length_imag`

### CODE REVIEW

File: EasyReflectometry/sample/elements/materials/material_mixture.py [hunk 1]
```diff
@@ -153,40 +153,19 @@ def _materials_constraints(self):
         iconstraint()
 
     @property
-    def fraction(self) -> Parameter:
```
Comment (andped10) [inline EasyReflectometry/sample/elements/materials/material_mixture.py:156]: These were duplicated

File: pyproject.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ build-backend = "hatchling.build"
 
 [project]
 name = "EasyReflectometryLib"
-version = "0.0.5"
+version = "0.0.5-dev"
```
Comment (rozyczko) [inline pyproject.toml:7]: why `-dev`? Do we use `-dev` anywhere?

**Review by rozyczko — APPROVED** [review]
Minor comments and questions raised, no issues with the code itself, so good to merge.

**Review by damskii9992 — APPROVED** [review]

**Review by damskii9992 — APPROVED** [review]
Looks good to me

File: EasyReflectometry/sample/assemblies/base_assembly.py [hunk 3]
```diff
@@ -175,3 +180,13 @@ def _disable_roughness_constraints(self):
...
+
+    def as_dict(self, skip: list = None) -> dict:
+        """Should produce a cleaned dict that matches the paramters in __init__
+
+        :param skip: List of keys to skip, defaults to `None`.
+        """
+        if skip is None:
+            skip = []
+        this_dict = super().as_dict(skip=skip)
+        return this_dict
```
[the remark this answers is not in the document]
Comment (andped10) [reply EasyReflectometry/sample/assemblies/base_assembly.py]: changed to code comment and consolidated in to base_core.py

File: EasyReflectometry/sample/elements/base_element.py [hunk 4]
```diff
@@ -38,3 +36,13 @@ def __repr__(self) -> str:
...
+
+    def as_dict(self, skip: list = None) -> dict:
+        """Should produce a cleaned dict that matches the paramters in __init__
+
+        :param skip: List of keys to skip, defaults to `None`.
+        """
+        if skip is None:
+            skip = []
+        this_dict = super().as_dict(skip=skip)
+        return this_dict
```
[the remark this answers is not in the document]
Comment (andped10) [reply EasyReflectometry/sample/elements/base_element.py]: same as above

File: .github/workflows/release-drafter.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  update_release_draft:
+    runs-on: ubuntu-latest
+    steps:
+      # Drafts your next Release notes as Pull Requests are merged into "main"
+      - uses: release-drafter/release-drafter@v5
```
[the remark this answers is not in the document]
Comment (andped10) [reply .github/workflows/release-drafter.yml]: changed

File: pyproject.toml (hunk 2 above)
Comment (andped10) [reply pyproject.toml]: Removed, left over from testing

**Pull request merged**